libs/market-list/src/lib/components/landing/select-market-list.tsx

```diff
@@ -0,0 +1,117 @@
+// Sparkline for the Market
+
+import { Arrow, Sparkline } from '@vegaprotocol/ui-toolkit';
+import { useRouter } from 'next/router';
+import { mapDataToMarketList } from '../../utils';
+import type { MarketList } from '../__generated__/MarketList';
+
+export interface MarketSparklineProps {
+  candles?: {
+    close: string | undefined | number;
+    open: string | undefined | number;
+  }[];
+}
+
+export const MarketSparkline = ({ candles }: MarketSparklineProps) => {
+  return (
+    <Sparkline
+      width={100}
+      height={20}
+      data={
+        candles
+          ?.filter((m) => !isNaN(Number(m.close)))
+          ?.map((m) => Number(m.close)) || []
+      }
+    />
+  );
+};
+
+// Select Market List
+
+export interface SelectMarketListProps {
+  data: MarketList | undefined;
+}
+
+export const SelectMarketList = ({ data }: SelectMarketListProps) => {
+  const thClassNames = (direction: 'left' | 'right') =>
+    `px-8 text-${direction} font-sans font-normal text-ui-small leading-9 capitalize mb-0 text-dark/80 dark:text-white/80`;
+  const tdClassNames =
+    'px-8 font-sans leading-9 capitalize text-ui-small text-right';
+  const priceChangeClassNames = (value: number) =>
+    value === 0
+      ? 'text-black dark:text-white'
+      : value > 0
+      ? `text-green`
+      : `text-red`;
+  const boldUnderlineClassNames =
+    'px-8 underline font-sans text-base leading-9 font-bold tracking-tight decoration-solid text-ui light:hover:text-black/80 dark:hover:text-white/80';
+
+  const { pathname, push } = useRouter();
+
+  return (
+    <div className="max-h-[40rem] overflow-x-auto">
+      <table className="relative h-full min-w-full whitespace-nowrap">
+        <thead className="sticky top-0 z-10 dark:bg-black bg-white">
+          <tr>
+            <th className={thClassNames('left')}>Market</th>
+            <th className={thClassNames('right')}>Last Price</th>
+            <th className={thClassNames('right')}>Change (24h)</th>
+            <th className={thClassNames('right')}></th>
```

Comment:
Yeah, no expiry date. I am not sure what is that supposed to be. I thought it was `market.marketTimestamps.close` but that is mostly null as it comes from our APIs. @mattrussell36 thinks it can be found in `market.tradableInstrument.instrument.product.oracleSpecForTradingTermination` but that is only for futures, and I couldn’t get the right data yet.

Reply:
the expiry date was in the design ![image](https://user-images.githubusercontent.com/16125548/169013898-da9cce59-f2ae-4592-ae70-82e4c15d3349.png)

Reply:
Yeah this was removed in a previous itteration of core, can we add a follow up ticket to look into it. I think futures markets without a termination time is probably something we can't really show come mainnet :+1: